package/MDAnalysis/analysis/base.py

```diff
@@ -40,18 +41,111 @@
 logger = logging.getLogger(__name__)
 
 
+class Results(dict):
+    r"""Container object for storing results.
+
+    Results are dictionaries that provide two ways in which can values be
+    accessed: by dict key ``results["value_key"]`` or by object attribute,
+    ``results.value_key``. They store all results obatined from an analysis
+    after the ``run`` call.
+
+    The current implementation is similar to the :class:`sklearn.utils.Bunch`
+    class in `scikit-learn`_.
+
+    .. _`scikit-learn`: https://scikit-learn.org/
+
+
+    Examples
+    --------
+    >>> results = Results(a=1, b=2)
+    >>> results['b']
+    2
+    >>> results.b
+    2
+    >>> results.a = 3
+    >>> results['a']
+    3
+    >>> results.c = [1, 2, 3, 4]
+    >>> results['c']
+    [1, 2, 3, 4]
+
+
+    Raises
+    ------
+    ValueError
+        If a to assigned attribute has the same name as a default dictionary
```

Comment:
```suggestion
        If an assigned attribute has the same name as a default
```

? I'm not sure what this is referring to